planner/languages/nginx/nginx_planner.go

```diff
@@ -92,12 +79,35 @@ var startCommand = strings.TrimSpace(`
 `)
 
 const nginxShellStartScript = `
-shell-nginx = pkgs.writeShellScriptBin "shell-nginx" ''
+#!/bin/bash
+
+welcome() {
+	echo "##### WARNING: nginx planner is experimental #####
 
-echo "Starting nginx with command:"
-echo "nginx -p %[1]s -c %[2]s -e /tmp/error.log -g \"pid /tmp/mynginx.pid;daemon off;\""
-nginx -p %[1]s -c %[2]s -e /tmp/error.log -g "pid /tmp/shell-nginx.pid;daemon off;"
-'';`
+	You may need to add
+
+	\"include shell-helper-nginx.conf;\"
+
+	to your %[2]s file to ensure the server can start in the nix shell.
+
+	Use \"shell-nginx\" to start the server"
+}
+
+pkg_path=$(readlink -f $(which nginx) | sed -r "s/\/bin\/nginx//g")
+conf_path=$pkg_path/conf
+
+mkdir -p %[1]s/.devbox/gen/nginx
+ln -sf $conf_path/* %[1]s/.devbox/gen/nginx/
+ln -sf $(pwd)/%[1]s/.devbox/gen/shell-helper-nginx.conf %[1]s/.devbox/gen/nginx/shell-helper-nginx.conf
+for file in %[1]s/*; do if [[ ! $file = .devbox ]]; then ln -sf $(pwd)/%[1]s/$file .devbox/gen/nginx/$file; fi; done
+
+shell-nginx() {
+	echo "Starting nginx with command:"
+	echo "nginx -p %[1]s -c %[1]s/.devbox/gen/nginx/%[2]s -e /tmp/error.log -g \"pid /tmp/mynginx.pid;daemon off;\""
+	nginx -p %[1]s -c %[1]s/.devbox/gen/nginx/%[2]s -e /tmp/error.log -g "pid /tmp/shell-nginx.pid;daemon off;"
+}
+welcome
```

Comment:
This is long enough that is could benefit from being in own `sh` file and we can embed it.